js/data-frame/index.tsx

```diff
@@ -1018,6 +1019,21 @@ const ShinyDataGrid: FC<ShinyDataGridProps<unknown>> = ({
   );
 };
 
+/**
+ * Solves the problem arising when there's more than one column with the same name (or when
+ * the column name is illegal, like in the case of an empty name) by returning a name with
+ * enough spaces appended so that it becomes unique.
+ * @param columns Names of existing columns.
+ * @param initName String to start the search with.
+ * @returns A string that does not exist in columns and thus can be used as column ID.
+ */
+function findUnusedColumnName(columns: ReadonlyArray<string>, initName = " "): string {
```

Comment:
Can we move `initName` into the function as this will only hold true if the `initName` is `" "`. 

If other column names can be expanded, then they could both resolve to the same `"   "` string.

However, if we know that `""` is the only one being expanded and there are no duplicate names, then this function is only called once... which is safe.

Changing to something more specific...
```suggestion
function emptyColumnNameToId(columns: ReadonlyArray<string>): string {
  let initName = " ";
```